File: java/src/main/java/vsdk/toolkit/common/statistics/PotentialStatistics.java
package vsdk.toolkit.common.statistics;

// Potential times area

public class PotentialStatistics {
    public double averageDirectPotential;
    public double maxDirectPotential;
    public double maxDirectImportance;
    public double totalDirectPotential;

    public PotentialStatistics() {
        averageDirectPotential = 0.0;
        maxDirectPotential = 0.0;
        maxDirectImportance = 0.0;
        totalDirectPotential = 0.0;
    }

    public void reset() {
        averageDirectPotential = 0.0;
        maxDirectPotential = 0.0;
        maxDirectImportance = 0.0;
        totalDirectPotential = 0.0;
    }
}

File: typescript/src/main/vsdk/toolkit/common/statistics/PotentialStatistics.ts
export class PotentialStatistics {
  public averageDirectPotential: number;
  public maxDirectPotential: number;
  public maxDirectImportance: number;
  public totalDirectPotential: number;

  public constructor() {
    this.averageDirectPotential = 0.0;
    this.maxDirectPotential = 0.0;
    this.maxDirectImportance = 0.0;
    this.totalDirectPotential = 0.0;
  }

  public reset(): void {
    this.averageDirectPotential = 0.0;
    this.maxDirectPotential = 0.0;
    this.maxDirectImportance = 0.0;
    this.totalDirectPotential = 0.0;
  }
}
